src/components/Application.ts

```diff
@@ -167,9 +173,25 @@ export const ApplicationFunction: ForwardRefRenderFunction<PixiApplication, Appl
         }
     }, [defaultTextStyle]);
 
+    // eslint-disable-next-line consistent-return
+    useEffect(() =>
```

Comment:
Is there ever a situation where a canvasElement is reused, so its root is unqueued from unmount, but then the old `oninit` is called from the prior render?

Reply:
If an `<Application>` is mounted, allowed to initialise, unmounted, then remounted and allowed to reinitialise, then the canvas would be reused and `onInit` would be called twice. However, React's strict mode shouldn't be able to cause that thanks to the unmount queue.